### ISSUE TITLE
Unhandled rejection Error: Error: File.contents can only be a Buffer, a Stream, or null.

### ISSUE THREAD

**Issue by yawlhead91** [addyosmani/critical#218]
Hi there great plugin, I know this issue has already been submitted but my setup is quite different. I have created my own private node module that uses your critical plugin. I call my node module like so passing in the options 

Gulp.coffee

```javascript
gulp.task 'critical_css', ->
    websiteTools.critical({
        inline: true,
        src: buildDir + "/**/*.html",
        dest: buildDir + "/**/*.html",
        css: ["./public/assets/css/style.min.css"],
        width: 1366,
        height: 768,
        inlineImages: true,
        minify: true,
        extract: true,
        ignore: ["@font-face"],
        timeout:60000
    }).then (res) ->
        console.log "Response", res
    .catch (err) ->
        console.log "Error: ", err
    return
```

Then I have my module that finds all the files and recursively runs critical like so.

```javascript 

#! /usr/bin/env node
/* jshint esversion: 6 */
require('events').EventEmitter.prototype._maxListeners = 100;

const critical = require('critical');
const glob = require('glob');
const Spinner = require('cli-spinner').Spinner;
const path = require("path");
const fs = require("fs");

// Spinner to show users progress
var spinner = new Spinner('Generating critical css path ... %s');
spinner.setSpinnerString('|/-\\');

/*
 * Return if files are present
 * @param {callback(error, array)} callback to return error or files
 * @return {} none
 */
function findFiles(options) {
    return new Promise(function (resolve, reject) {
        var files = glob.sync(options.src, []);
        if (files.length) {
            return resolve(files);
        }
        return reject(new Error('No files found'));
    });
}

/*
 * Generate critical css
 * @param {array, callback} array of file paths
 * @return {callback(error)}
 */
function applyCritical(files, options, index) {
    // Get current file
    var file = files[index];
    var config = options;

    var filePath = path.join(process.cwd(), file)

    if (!fs.existsSync(filePath)) {
        console.log("File does not exist: ", file);
        return
    }

    config.src = filePath;
    config.dest = filePath;

    if (index >= files.length) {
        return true;
    }

    critical.generate(config).then(() => {
        return applyCritical(files, options, index + 1);
    }).catch(error => {
        throw new Error(error);
    });
}

/**
 *
 *
 */
module.exports = options => {
    return new Promise(function (resolve, reject) {

        if (!options) {
            throw new Error('Required options not provided');
        }

        spinner.start();

        findFiles(options).then(files => {
            return applyCritical(files, options, 0);

        }).then(res => {
            spinner.stop(true); // Stop spinner
            return resolve(res);

        }).catch(error => {
            spinner.stop(true); // Stop spinner
            return reject(error);
        });

    });
};

```

But I keep getting this error

```
Unhandled rejection Error: Error: File.contents can only be a Buffer, a Stream, or null.
    at critical.generate.then.catch.error (/Users/alexkirwan/Development/website-teamworkdesk.com/node_modules/@teamwork/website-tools/lib/critical.js:59:11)
    at tryCatcher (/Users/alexkirwan/Development/website-teamworkdesk.com/node_modules/bluebird/js/release/util.js:16:23)
    at Promise._settlePromiseFromHandler (/Users/alexkirwan/Development/website-teamworkdesk.com/node_modules/bluebird/js/release/promise.js:512:31)
    at Promise._settlePromise (/Users/alexkirwan/Development/website-teamworkdesk.com/node_modules/bluebird/js/release/promise.js:569:18)
    at Promise._settlePromise0 (/Users/alexkirwan/Development/website-teamworkdesk.com/node_modules/bluebird/js/release/promise.js:614:10)
    at Promise._settlePromises (/Users/alexkirwan/Development/website-teamworkdesk.com/node_modules/bluebird/js/release/promise.js:689:18)
    at Async._drainQueue (/Users/alexkirwan/Development/website-teamworkdesk.com/node_modules/bluebird/js/release/async.js:133:16)
    at Async._drainQueues (/Users/alexkirwan/Development/website-teamworkdesk.com/node_modules/bluebird/js/release/async.js:143:10)
    at Immediate.Async.drainQueues (/Users/alexkirwan/Development/website-teamworkdesk.com/node_modules/bluebird/js/release/async.js:17:14)
    at runCallback (timers.js:672:20)
    at tryOnImmediate (timers.js:645:5)
    at processImmediate [as _immediateCallback] (timers.js:617:5)
```

I have tried all different versions of the module and npm, node, adding, changing base paths ect still no joy. Any idea why I am getting this. Thanks

**Comment by yawlhead91** [comment]
@addyosmani hey I figured out the issue, if you see here at 

https://github.com/addyosmani/critical/blob/master/lib/file-helper.js#L329-L343

The issue is on line 340 , I just made a small change here , it was trying to assign a array as the content as opposed to the buffer inside. I can do a quick pull request if you wouldnt mind. 

```javascript
return fs.readFileAsync(file.path).then(function (contents) {
      file.contents = contents.pop();
      return file;
});
```
By the way just finished your javascript patterns book great work.

**Comment by bezoerb** [comment]
Hey @yawlhead91 thanks for reporting this. A pull request along with a test showing that your fix works would be awesome 😊

**Comment by bezoerb** [comment]
May be related https://github.com/addyosmani/critical/issues/191

**Comment by yawlhead91** [comment]
@bezoerb I would say that they are related, the only thing that both these cases have in common is gulp.  I am going to spend a few hours today debugging this issue in depth, and make some test reproduce the issues.

**Comment by iandevlin** [comment]
Seeing this issue too on my first attempt at using critical, would love to see a fix.

**Comment by bezoerb** [comment]
@yawlhead91 any updates on this?

**Comment by bezoerb** [comment]
Should be fixed in [v1.0.0](https://github.com/addyosmani/critical/releases/tag/v1.1.0). Feel free to reopen if the error still exists.

**Comment by dylanspurgin** [comment]
I am still seeing this error using v1.0.0. Everything is working (very slowly) if I downgrade to 0.8.0.
